sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/RestrictionTracker.java

```diff
@@ -80,6 +81,7 @@
    *     represent based upon the current known remaining amount of work.
    * @return a {@link SplitResult} if a split was possible, otherwise returns {@code null}.
```

Comment:
We should make it clear that a checkpoint must return a valid split result or null where null represents that there is no residual.